docs/source/topics/experimental.rst

```diff
@@ -0,0 +1,87 @@
+Experimental APIs
+=================
+
+Chalice maintains backwards compatibility for all features that appear in this
+documentation.  Any Chalice application using version 1.x will continue to work
+for all future versions of 1.x.
+
+We also believe that Chalice has a lot of potential for new ideas and APIs,
+many of which will take several iterations to get right.  We may implement a
+new idea and need to make changes based on customer usage and feedback.  This
+may include backwards incompatible changes all the way up to the removal of
+a feature.
+
+To accommodate these new features, Chalice has support for experimental APIs,
+which are features that are added to Chalice on a provisional basis.  Because
+these features may include backwards incompatible changes, you must explicitly
+opt-in to using these features.  This makes it clear that you are using an
+experimental feature that may change.
+
+Opting-in to Experimental APIs
+------------------------------
+
+Each experimental feature in chalice has a name associated with it.  To opt-in
+to an experimental API, you must have the feature name to the
+``experimental_feature_flags`` attribute on your ``app`` object.
+This attribute's type is a set of strings.
+
+.. code-block:: python
+
+    from chalice import Chalice
+
+    app = Chalice('myapp')
+    app.experimental_feature_flags.update([
+        'MYFEATURE1',
+        'MYFEATURE2',
+    ])
+
+
+If you use an experimental API without opting-in, you will receive
+a message whenever you run a Chalice CLI command.  The error message
+tells you which feature flags you need to add::
+
+    $ chalice deploy
+    You are using experimental features without explicitly opting in.
+    Experimental features do not guarantee backwards compatibility and may be removed in the future.
+    If you still like to use these experimental features, you can opt-in by adding this to your app.py file:
+
+    app.experimental_feature_flags.update([
+        'BLUEPRINTS'
+    ])
+
+
+    See https://chalice.readthedocs.io/en/latest/topics/experimental.rst for more details.
+
+The feature flag only happens when running CLI commands.  There are no runtime
+checks for experimental features once your application is deployed.
+
+
+List of Experimental APIs
+=========================
+
+In the table below, the "Feature Flag Name" column is the value you
+must add to the ``app.experimental_feature_flags`` attribute.
+The status of an experimental API can be:
+
+* ``Trial`` - You must explicitly opt-in to use this feature.
```

Comment:
Could we add a column here for a link to context. PR/Feature request thread or something like that. Especially useful for tracking when/why something was rejected/accepted.

Reply:
+1